go/database/mpt/proof.go

```diff
@@ -93,43 +94,68 @@ func (p WitnessProof) IsValid() bool {
 }
 
 // GetAccountInfo extracts an account info from the witness proof for the input root hash and the address.
-// If the witness proof contains an account for the input address, it returns its information.
-// If the proof does not contain an account, it returns false.
+// If the proof was complete, this method returns true, otherwise it returns false.
+// The proof was complete if it could fully determine either existence or non-existence of the account
+// in the proof. In other words, it was possible to reach either an account node or an empty node.
```

Comment:
nit: I think this describes implementation details not really relevant for the user of this API.

The essential thing is:
- the `bool` is true => the returned info is proven and can be used
- the `bool` is false or an error is produced => the returned info must not be used

Reply:
ok, make sense, updated